application/templates/base.html

```diff
@@ -50,47 +59,75 @@
           ga('require', 'eventTracker', {
             attributePrefix: 'data-'
           });
-          {% block google_analytics %}{% endblock %}
+          {% block googleAnalytics %}{% endblock %}
           ga('send', 'pageview');
         </script>
         <script src="/static/vendor/autotrack/autotrack.js"></script>
     {% endif %}
 
-    {% block endhead %} {% endblock %}
-
+    {% block headEnd %}
+      {% if not static_mode %}
+        <link rel="stylesheet" href="/static/stylesheets/{{ 'cms.css' | version_filter }}">
+      {% endif %}
+    {% endblock %}
   </head>
-  <body class="{% block body_classes %}{% endblock %}">
-    <script>document.body.className = ((document.body.className) ? document.body.className + ' js-enabled' : 'js-enabled');</script>
+  
+  <body class="{% block bodyClasses %}{% endblock %}">
+    {% block bodyStart %}
+      <script>document.body.className = ((document.body.className) ? document.body.className + ' js-enabled' : 'js-enabled');</script>
+    {% endblock %}
 
+    {% block skipLink %}
     <div id="skiplink-container">
       <div>
         <a href="#content" class="skiplink">{{ skip_link_message|default('Skip to main content') }}</a>
       </div>
     </div>
+    {% endblock %}
 
-    {% if not static_mode %}
-      {% include 'static_site/_preview.html' %}
-    {% endif %}
-
+    {% block cookieMessage %}
     <div id="global-cookie-message">
       <div class="cookie-message full-width">
         <p>GOV.UK uses cookies to make the site simpler. <a href="/cookies">Find out more about cookies</a></p>
       </div>
     </div>
+    {% endblock %}
 
-    {% include '_shared/_header.html' %}
-
-    {% block after_header %}{% endblock %}
+    {% block header %}
+      {% if not static_mode %}
+        {% include 'static_site/_preview.html' %}
+      {% endif %}
+      
+      {% include '_shared/_header.html' %}
+    {% endblock %}
 
-    <div id="global-header-bar"></div>
+    <div id="global-header-bar"></div> {# This is a hook into CSS that adds a blue bar beneath the top black navbar #}
 
     {% include '_shared/_phase_banner.html' %}
 
-    {% block main_content %}{% endblock %}
+    {% block main %}
+      {% block mainElementDefinition %}<main id="content" class="main">{% endblock %}
+        
+      {% block beforeContent %}
+        {% block breadcrumbs %}
+          {{ render_breadcrumbs(breadcrumbs) }}
```

Comment:
It feels a _bit_ weird to both define a `breadcrumbs` block AND for the site template to assume that a `breadcrumbs` variable will be set in every template.

Could we wrap the `render_breadcrumbs` method in a condition which checks whether `breadcrumbs` is defined, to avoid having to set it to `none` on pages without breadcrumbs?

Reply:
This check already occurs in the `render_breadcrumbs` macro - so maybe we can just drop the block altogether.

Reply:
👍